### PR TITLE
[Elixir] Adds concepts to bracket-push exercise

### PR DESCRIPTION
**Pull request by fireproofsocks**
My 2nd PR (hope I'm doing this correctly)...
PR for Issue #422

### CODE REVIEW

**Comment by neenjaw** [comment]
I think an important version is the stack-based implementation using a list, but I'm not sure it flushes out more concepts than what we already have so far with this process.  So not sure if it's important to go through it for that

File: languages/elixir/reference/exercise-concepts/bracket-push.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    - boolean
+- pattern matching
+  - on function signature
+- guards
+  - when
+- standard library modules
+  - Kernel
+    - in/2
+- types
+  - binaries
```
Comment (neenjaw) [inline languages/elixir/reference/exercise-concepts/bracket-push.md]:
```suggestion
  - binaries 
      - special forms `<< >>` syntax
```

File: languages/elixir/reference/exercise-concepts/bracket-push.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+- naming
+- modules
+- module attributes
+  - @doc
+    - documentation
+  - @spec
+    - types
+    - String.t()
+    - boolean
+- pattern matching
```
Comment (neenjaw) [inline languages/elixir/reference/exercise-concepts/bracket-push.md]:
```suggestion
- pattern matching
  - in binary special forms
```

Comment (fireproofsocks) [reply languages/elixir/reference/exercise-concepts/bracket-push.md]: thanks for the clarification!

**Comment by fireproofsocks** [comment]
I haven't dug through the solutions on the site... but I'm happy to include a code sample if we have one of the "stack-based implementation using a list"...

**Comment by Cohen-Carlisle** [comment]
Here is my solution with uses a list as a stack to check the brackets: https://exercism.io/tracks/elixir/exercises/matching-brackets/solutions/fc3bb41f32ae4ce99d7fc875db4ada94

File: languages/elixir/reference/exercise-concepts/bracket-push.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  defp seek_end(_, <<h :: binary - size(1), _ :: binary>>) when h in @closing_delimiters,
+       do: {:error, "Unmatched closing delimiter: '#{h}'"}
+
+  # Filler character --> shift and keep seeking
+  defp seek_end(closing_delimiter, <<_ :: binary - size(1), tail :: binary>>),
+       do: seek_end(closing_delimiter, tail)
+
+end
+```
+
```
Comment (neenjaw) [inline languages/elixir/reference/exercise-concepts/bracket-push.md]: Could add in the stack-based solution here

**Comment by neenjaw** [comment]
Close #422 when merged

**Comment by neenjaw** [comment]
Hey @fireproofsocks, hope it's okay that I added @Cohen-Carlisle's solution to your file to this.

Can I merge?

**Pull request merged**